src/Common/src/System/IO/StringParser.cs

```diff
@@ -171,17 +171,12 @@ public unsafe int ParseNextInt32()
                     {
                         ThrowForInvalidData();
                     }
-                    result = checked((result * 10) + d);
+                    result = negative ? checked((result * 10) - d) : checked((result * 10) + d);
```

Comment:
Why this change?  Rather than deleting the below section, could we instead just change that to:
```C#
if (negative)
{
    result = checked(result * -1);
}
```
?